src/graphql/schema/collection/createCollectionGqlType.ts

```diff
@@ -37,6 +37,7 @@ export default function createCollectionGqlType<TValue> (
         primaryKey && [options.nodeIdFieldName, {
           description: 'A globally unique identifier. Can be used in various places throughout the system to identify this single value.',
           type: new GraphQLNonNull(GraphQLID),
+          externalFieldNameDependencies: primaryKey._keyTypeFields.map(([fieldName, field]) => field.externalFieldName),
```

Comment:
I didn't know where to grab this from, so just grabbed the nearest available thing. Also what if it's a compound primary key?